src/prefect/storage/bitbucket.py

```diff
@@ -38,35 +40,47 @@ class Bitbucket(Storage):
     - Call `prefect register -f flow.py` to register this flow with Bitbucket storage.
 
     Args:
-        - project (str): Project that the repository will be in.  Not equivalent to a GitHub project;
+        - project (str): project that the repository will be in.  Not equivalent to a GitHub project;
             required value for all Bitbucket repositories.
         - repo (str): the repository name, with complete taxonomy.
-        - host (str, optional): If using Bitbucket server, the server host. If not specified, defaults
-            to Bitbucket cloud.
+        - workspace (str, optional): the workspace name. Bitbucket cloud only.
+        - host (str, optional): the server host. Bitbucket server only.
         - path (str, optional): a path pointing to a flow file in the repo
-        - ref (str, optional): a commit SHA-1 value or branch name
-        - access_token_secret (str, optional): The name of a Prefect secret
+        - ref (str, optional): a commit SHA-1 value or branch name or tag. If not specified,
+            defaults to master branch for the repo.
+        - access_token_secret (str, optional): the name of a Prefect secret
             that contains a Bitbucket access token to use when loading flows from
-            this storage.
+            this storage. Bitbucket Server only
+        - cloud_username_secret (str, optional): the name of a Prefect secret that contains a
+            Bitbucket username to use when loading flows from this storage. Bitbucket Cloud only.
+        - cloud_app_password_secret (str, optional): the name of a Prefect secret that contains a
+            Bitbucket app password, from the account associated with the `cloud_username_secret`,
+            to use when loading flows from this storage. Bitbucket Cloud only.
         - **kwargs (Any, optional): any additional `Storage` initialization options
     """
 
     def __init__(
         self,
         project: str,
         repo: str,
+        workspace: str = None,
         host: str = None,
         path: str = None,
         ref: str = None,
         access_token_secret: str = None,
+        cloud_username_secret: str = None,
+        cloud_app_password_secret: str = None,
         **kwargs: Any,
     ) -> None:
         self.project = project
         self.repo = repo
+        self.workspace = workspace
         self.host = host
         self.path = path
         self.ref = ref
         self.access_token_secret = access_token_secret
+        self.cloud_username_secret = cloud_username_secret
+        self.cloud_app_password_secret = cloud_app_password_secret
```

Comment:
Could you add a warning if these are set and workspace is not and vice-versa? Basically

- if workspace is set but the cloud auth is not
- if workspace is not set but the cloud auth is

with the goal of indicating to the user which fields they'll need to set for things to work. If authentication is not _required_ for BitbucketCloud then nevermind. We may want to raise an exception instead of a warning if it'll always fail at runtime.

Reply:
Ah after getting further down I see these will be loaded from the environment if not set :D 

I would change this to

- if self.workspace is set and self.access_token_secret is set _then_ throw an exception saying they've set the wrong secret variable